### ISSUE TITLE
Regression when using dind.enabled for volumes mounts on GKE

### ISSUE THREAD

**Issue by zapo** [buildkite/charts#75]
**Is this a request for help?**:
Yes
---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):
BUG REPORT, likely reopen/duplicate of https://github.com/buildkite/charts/issues/66

**Version of Helm and Kubernetes**:
```
15:48 $ helm version
version.BuildInfo{Version:"v3.2.4", GitCommit:"0ad800ef43d3b826f31a5ad8dfbb4fe05d143688", GitTreeState:"clean", GoVersion:"go1.14.4"}
```

Cluster version: 1.15.12-gke.2

**Which chart**:
```
15:51 $ helm history buildkite-agent
REVISION        UPDATED                         STATUS          CHART           APP VERSION     DESCRIPTION
1               Tue Aug 18 14:39:08 2020        deployed        agent-0.4.4     3.22.1          Install complete
```

**What happened**:

It seems like since https://github.com/buildkite/charts/pull/70 got merged that any volume mounted with `docker run` from /var/buildkite/lib is only exposing empty directories. This is similar problem as the one that was flagged in https://github.com/buildkite/charts/issues/66 but it seems like it regressed after #70 got merged. 

Manually updating the deployment to set mountPath back to `/var/buildkite/lib` instead of `/var/buildkite/shared` seems to fix the issue even-though I'm not sure why/how things work.

This can be tested when used alongside https://github.com/buildkite-plugins/docker-buildkite-plugin. The mounted checked out volume from /var/buildkite/lib is empty. Inspecting it on host properly shows the underlying folders and the checked out code, but not from within the container.

**What you expected to happen**:

To be able to interact with checked out code in steps defined using https://github.com/buildkite-plugins/docker-buildkite-plugin.

**How to reproduce it** (as minimally and precisely as possible):

```
helm install buildkite-agent buildkite/agent --set agent.token="..." --set privateSshKey="..." --set dind.enabled=true
```

trigger a manual build with something like:

```
steps:
  - command:
      - "ls -la /workdir"
    plugins:
      - docker#v3.5.0:
          image: "alpine:latest"
          mount-buildkite-agent: false
          volumes:
            - /var/run/docker.sock:/var/run/docker.sock
```
Notice that the workdir folder is empty.

**Anything else we need to know**:

It seems important that a build ran at least once so the /var/buildkite/lib folder exists in the container.
I tried to reproduce the volume mount the plugin does directly into the agent with:

```
kubectl exec -it buildkite-agent-agent-... -c agent -- bash
```
then
```
bash-4.4# docker run -v /var/buildkite:/workdir alpine ls -la /workdir
total 12
drwxrwxrwx    3 root     root          4096 Aug 19 00:07 .
drwxr-xr-x    1 root     root          4096 Aug 19 00:14 ..
drwxr-xr-x    3 root     root          4096 Aug 19 00:07 builds
```

and I can see the folder tree, but they all appear empty even-though they are populated on host!

**Comment by O1ahmad** [comment]
Hey @zapo, a regression for sure. I submitted a reversion not long after the commit but looks like it was [botched](https://github.com/buildkite/charts/commit/e3e48fb6bab0ae55e2160d0a48e6a83bad581a3c#diff-f75fe9b5059d01dedd51e6a9d047fb11) :disappointed:

The change is in my fork: https://github.com/0x0I/charts/blob/master/stable/agent/templates/deployment.yaml#L93. Feel free to submit a fix or I can - whatever works.

**Comment by zapo** [comment]
@0x0I Thanks for the quick answer! Tentative fix in https://github.com/buildkite/charts/pull/76
